esrally/track/params.py

```diff
@@ -344,10 +408,14 @@ def __init__(self, track, params, **kwargs):
         super().__init__(track, params, **kwargs)
         if len(track.indices) == 1:
             default_index = track.indices[0].name
+        elif len(track.data_streams) == 1:
+            default_index = track.data_streams[0].name
```

Comment:
I think a more generic name, such as `default_target` would be less confusing. This applies similarly also to other parts of this PR such as `ForceMergeParamSource`.

Reply:
Agreed. Should also prevent the use of index when data streams are in use and vise versa? This would be consistent with our proposal for the track i.e. `data-streams` and `indices` cannot co-exist.